### PR TITLE
feat: changed deployment

### PR DESCRIPTION
**Pull request by janrtvld**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -2,29 +2,83 @@ version: '3.5'
...
+          - node.role == worker
+      labels:
+        # Enable traefik so the container can be exposed to the outside world
+        traefik.enable: 'true'
+
+        traefik.http.routers.client-demo.rule: Host(`demo.animo.id`)
+        traefik.http.routers.client-demo.entrypoints: web-secure
+        traefik.http.routers.client-demo.tls.certresolver: zerossl
+        traefik.http.routers.client-demo.service: client-demo-service
+        traefik.http.routers.client-demo.middlewares: chain-oauth@file
```
Comment (TimoGlastra) [inline docker-compose.yml]: This will require you to sign in, we don't need that for the demo

File: config/afj-prod.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (TimoGlastra) [inline config/afj-prod.json]: The server already configures it's own config right? This is only needed if you use the afj-rest cli without importing the startServer etc... method yourself.

File: docker-compose.yml [hunk 3]
```diff
@@ -2,29 +2,83 @@ version: '3.5'
...
-      AGENT_PUBLIC_DID_SEED: ${AGENT_PUBLIC_DID_SEED}
+      AGENT_WALLET_KEY: ${AGENT_WALLET_KEY}
       AGENT_ENDPOINT: https://didcomm.demo.animo.id
-    ports:
-      - 5000
-      - 5001
+      AGENT_PUBLIC_DID_SEED: ${AGENT_PUBLIC_DID_SEED}
+
+    volumes:
+      - indy_client:/root/.indy_client
```
Comment (TimoGlastra) [inline docker-compose.yml]: volumes with docker stack deploy don't work out of the box, I think we can just do without a volume for now

```suggestion
```

File: docker-compose.yml [hunk 4]
```diff
@@ -2,29 +2,83 @@ version: '3.5'
...
+
     networks:
-      - https-proxy
+      - traefik
+
+    ports:
+      - '5000'
+      - '5001'
+
+    command: --config /config.json
```
Comment (TimoGlastra) [inline docker-compose.yml]: The server already has it's own entrypoint I think, so we can remove this
```suggestion
```

File: docker-compose.yml [hunk 5]
```diff
@@ -2,29 +2,83 @@ version: '3.5'
...
-      - 5000
-      - 5001
+      AGENT_PUBLIC_DID_SEED: ${AGENT_PUBLIC_DID_SEED}
+
+    volumes:
+      - indy_client:/root/.indy_client
+
+    configs:
+      - source: afj_config
+        target: /config.json
```
Comment (TimoGlastra) [inline docker-compose.yml]: Not needed with already existing config in server
```suggestion
```

File: docker-compose.yml [hunk 6]
```diff
@@ -2,29 +2,83 @@ version: '3.5'
...
+    networks:
+      - traefik
+
+volumes:
+  indy_client:
+
+configs:
+  afj_config:
+    external: true
+    name: demo_afj_rest_config_${AFJ_CONFIG_ID}
```
Comment (TimoGlastra) [inline docker-compose.yml]:
```suggestion
```

File: docker-compose.yml [hunk 7]
```diff
@@ -29,37 +29,22 @@ services:
...
         constraints:
           - node.role == worker
       labels:
-        # Enable traefik so the container can be exposed to the outside world
         traefik.enable: 'true'
-
         traefik.http.routers.client-demo.rule: Host(`demo.animo.id`)
-        traefik.http.routers.client-demo.entrypoints: web-secure
-        traefik.http.routers.client-demo.tls.certresolver: zerossl
-        traefik.http.routers.client-demo.service: client-demo-service
```
Comment (TimoGlastra) [inline docker-compose.yml]: These are needed @janrtvld, just the middleware is not needed!

Comment (TimoGlastra) [reply docker-compose.yml]: Sorry should have been more clear

Comment (janrtvld) [reply docker-compose.yml]: Ye just fixed it thanks

**Pull request merged**